### ISSUE TITLE
Incorrect env resolution when using $(OTHER_ENV_VAR) syntax

### ISSUE THREAD

**Issue by silvajp2** [telepresenceio/telepresence#1900]
When using `telepresence --namespace staging --swap-deployment instance --run bash`

On a deployment which env contains:
```yaml
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: PREFIX
          value: $(POD_NAMESPACE)/v1/io/
```
Telepresence incorrectly escapes `$(POD_NAMESPACE)` instead of resolving it to the value of POD_NAMESPACE.

On the `traffic-agent` I got:
```bash
export TEL_APP_PREFIX='$(POD_NAMESPACE)/v1/io/'
export TEL_APP_POD_NAMESPACE='staging'
```

Instead of the correct:
```bash
export TEL_APP_PREFIX='staging/v1/io/'
export TEL_APP_POD_NAMESPACE='staging'
```

```bash
$ telepresence status 
Root Daemon: Running
  Version     : v2.3.4 (api 3)
  Manager Port: 0
 ....
```

**Comment by vvazharksi** [comment]
Same with status.hostIP. Any idea ?

**Comment by thallgren** [comment]
@vvazharksi thanks for reporting this, and yep, I have a pretty good idea. See linked PR.

**Comment by efunk** [comment]
This was released in Telepresence 2.4.0. Please see the release notes: https://www.getambassador.io/docs/telepresence/latest/release-notes
